pkg/services/dns/dns.go

```diff
@@ -168,10 +168,23 @@ func (s *Server) Mux() http.Handler {
 			return
 		}
 
-		s.handler.zonesLock.Lock()
-		s.handler.zones = append([]types.Zone{req}, s.handler.zones...)
-		s.handler.zonesLock.Unlock()
+		s.addZone(req)
 		w.WriteHeader(http.StatusOK)
 	})
 	return mux
 }
+
+func (s *Server) addZone(req types.Zone) {
+	s.handler.zonesLock.Lock()
+	for i, zone := range s.handler.zones {
+		if zone.Name == req.Name {
+			req.Records = append(req.Records, zone.Records...)
+			merge := append([]types.Zone{req}, s.handler.zones[i+1:]...)
+			s.handler.zones = append(s.handler.zones[:i], merge...)
```

Comment:
I think this would be equivalent to `s.handler.zones[i] = req` ? I've tested this builds, but I'm not 100% sure of golang's behaviour with such assignments.

Reply:
Done this change.